### PR TITLE
fix: make back button navigate one route up

### PR DESCRIPTION
**Pull request by dziraf**

### CODE REVIEW

**Review by ariansobczak-rst — APPROVED** [review]

**Comment by dziraf** [comment]
This fixes the issue where drawer doesn't close correctly, but introduces another issue where refreshing the page will result in 404. This is because `to=".." relative="path"` only removes URL's pathname up to last `/` which in case of `show` action removes record's ID which results in drawer getting closed. AdminJS doesn't have a record route without `:recordId` param defined though, so when you refresh the page you see a `CANNOT GET ...` error.

We'd probably have to rewrite routing so that routes are nested, this way we could simply use `relative="route"`.

**Comment by vinu** [comment]
Hmm but will this work? with filters? 

I just tried like this

```jsx
<StyledLink
      size="icon"
      onClick={() => navigate(-1)}
      rounded
      mr="lg"
      type="button"
    >
```
Seems working

**Comment by vinu** [comment]
nah - disregard that - it won't work with multiple show click, - the back click just loads the previous show action

**Comment by dziraf** [comment]
I rewrote routing so that back button works as expected now, but it turned out we also have to rewrite the Drawer component since it doesn't show list underneath anymore.


https://github.com/SoftwareBrothers/adminjs/assets/16668924/d8e1b0cd-a044-4934-8c6a-c771c9de44f5

**Comment by dziraf** [comment]
It looks to be working correctly now

https://github.com/SoftwareBrothers/adminjs/assets/16668924/63b9ceda-58d3-41b4-9d36-4ff67c5d1024

File: src/frontend/interfaces/action/build-action-click-handler.ts [hunk 1]
```diff
@@ -65,7 +67,17 @@ export const buildActionClickHandler = (
 
     // Default behaviour - you're navigated to action URL and logic is performed on it's route
     if (href) {
-      navigate(href)
+      const url = new URL(`relative:${href}`)
```
Comment (dziraf) [inline src/frontend/interfaces/action/build-action-click-handler.ts:70]: appending pathname with `<string>:` is a hack which allows us to use `new URL(...)` if we only have a relative url

Comment (ariansobczak-rst) [reply src/frontend/interfaces/action/build-action-click-handler.ts:70]: 👍🏻 Nice, I didn't know that.

**Review by ariansobczak-rst — APPROVED** [review]

**Comment by dziraf** [comment]
fixes https://github.com/SoftwareBrothers/adminjs/issues/1543

**Pull request merged**